mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/loader/MassIndexingThreadContext.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.massindexing.loader;
+
+/**
+ * Contextual information about a mass indexing proccess.
+ * @param <O> The options for mass indexing proccess.
+ */
+public interface MassIndexingThreadContext<O> {
+
+	/**
+	 * @return The mass options for loading process.
+	 */
+	O options();
+
+	/**
+	 * Get the in progress indexing process,
+	 *
+	 * @return true if process is active.
+	 */
+	boolean active();
```

Comment:
It is difficult to predict what will be in the interception in programming aspects of entity loading. Interception should be able to break entity load loops. By default, there is no obligation to use this because the first element of breaking the loop is checking for "Thread.interrupted ()". This is just an interception aid as if the programmer building the interception could influence what happens in the entity load loop, no matter which load strategy the interception was used for. In fact, it has the opposite effect for the entity loading loop. The entity load loop is not dependent on interception, it only checks the status of the active() function; 

Thanks to this, you can build standard entity loaders without worrying about what will happen in interception: 
```java
public class JavaBeanMapIndexingStrategy<E> implements MassIndexingEntityLoadingStrategy<E, JavaBeanIndexingOptions> {

............
	@Override
	public EntityIdentifierScroll createIdentifierScroll(MassIndexingThreadContext<JavaBeanIndexingOptions> context, Set<Class<? extends E>> includedTypes) throws InterruptedException {
		Set<?> identifiers = source.entrySet().stream()
				.filter( ent -> context.commonSuperType().isAssignableFrom( ent.getValue().getClass() ) )
				.filter( ent -> context.indexed( ent.getValue() ) )
				.map( Entry::getKey ).collect( Collectors.toSet() );
		Iterator<?> iterator = identifiers.iterator();
		return new EntityIdentifierScroll() {
........
			@Override
			public List<?> next() {
				int batchSize = context.options().batchSize();

				List<Object> destination = new ArrayList<>( batchSize );
				while ( iterator.hasNext() ) {
					destination.add( iterator.next() );

                                        // Checked condition, broken loop if requested by one of the interpretations.  
					if ( !context.active() ) {
						throw log.contextNotActiveWhileProducingIdsForBatchIndexing( context.includedEntityNames() );
					}

					if ( destination.size() == batchSize ) {
						return destination;
					}
				}
				return destination;
			}
		};
	}

	@Override
	public EntityLoader<E> createLoader(MassIndexingThreadContext<JavaBeanIndexingOptions> context, Set<Class<? extends E>> includedTypes) throws InterruptedException {
		return (identifiers) -> identifiers.stream().map( source::get ).collect( Collectors.toList() );

	}

}
```

Reply:
This can be replaced by using interception contexts. I will remove the built-in activity testing features.